compiler/rustc_mir_transform/src/simplify.rs

```diff
@@ -632,6 +650,16 @@ impl<'tcx> Visitor<'tcx> for UsedLocals {
             self.use_count[local] -= 1;
         }
     }
+
+    fn visit_var_debug_info(&mut self, var_debug_info: &VarDebugInfo<'tcx>) {
+        if let VarDebugInfoContents::Place(..) = var_debug_info.value
```

Comment:
Now that we have SROA in MIR, we may need to check `VarDebugInfoContents::Composite` too.

Reply:
For now I'm counting any debuginfo that's not just a direct mention of a local as a use of the variable that keeps it from being removed -- so my expectation is that I don't need to handle `Composite`, as it'll always end up calling `super_var_debug_info`, which will add a use for each local.

Any guesses at what might be wrong here, @cjgillot?  It's still failing in mipsel but [working](https://github.com/rust-lang/rust/actions/runs/5363521273/jobs/9731157734) in the normal  build, so I'm very puzzled.

I saw you mention https://github.com/rust-lang/rust/pull/109025#discussion_r1239491410 that you're waiting on this, but I'm unsure how to get it into a good place right now.